x-pack/plugins/security_solution/public/common/components/event_details/insights/related_cases.tsx

```diff
@@ -33,27 +35,47 @@ export const RelatedCases = React.memo<Props>(({ eventId }) => {
   const [relatedCases, setRelatedCases] = useState<RelatedCaseList | undefined>(undefined);
   const [hasError, setHasError] = useState<boolean>(false);
 
+  const { activeStep, isTourShown } = useTourContext();
+  const isTourActive = useMemo(
+    () => activeStep === AlertsCasesTourSteps.viewCase && isTourShown(SecurityStepId.alertsCases),
+    [activeStep, isTourShown]
+  );
   const renderContent = useCallback(() => renderCaseContent(relatedCases), [relatedCases]);
 
-  const getRelatedCases = useCallback(async () => {
-    let relatedCaseList: RelatedCaseList = [];
-    try {
-      if (eventId) {
-        relatedCaseList =
-          (await cases.api.getRelatedCases(eventId, {
-            owner: APP_ID,
-          })) ?? [];
-      }
-    } catch (error) {
-      setHasError(true);
-      toasts.addWarning(CASES_ERROR_TOAST(error));
+  const [shouldFetch, setShouldFetch] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!shouldFetch) {
+      return;
     }
-    setRelatedCases(relatedCaseList);
```

Comment:
The unit tests were yelling about this being called when the component was unmounted. Therefore, I refactored the async call to avoid this side effect